### PR TITLE
Fixes and docs for usage as plugin

### PR DESCRIPTION
**Pull request by nirs**
Using the new CommandDisplayNameAnnotation annotation introduced in Cobra 1.8.0.

Fixes help text issues with:
- help flag usage
- help command long description
- usage line for command without sub commands

### CODE REVIEW

File: site/content/user_guide.md [hunk 1]
```diff
@@ -748,3 +748,57 @@ Read more about it in [Shell Completions](completions/_index.md).
...
+Usage:
+  kubectl myplugin [command]
+
+Available Commands:
+  completion  Generate the autocompletion script for the specified shell
+  help        Help about any command
+  subcmd
+
+Flags:
+  -h, --help   help for kubectl-myplugin
```
Comment (nirs) [inline site/content/user_guide.md:801]: This revels a bug in the way we create the help for the root command. Working on a fix...

**Comment by nirs** [comment]
Example runs using https://github.com/nirs/kubectl-cobraplugin

Root command help:

```
$ kubectl cobraplugin
Usage:
  kubectl cobraplugin [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  subcmd      

Flags:
  -h, --help   help for kubectl cobraplugin

Use "kubectl cobraplugin [command] --help" for more information about a command.
```

Sub command help:

```
$ kubectl cobraplugin subcmd -h
Usage:
  kubectl cobraplugin subcmd [flags]

Flags:
  -h, --help   help for subcmd
```

Help command help:
```
$ kubectl cobraplugin help -h
Help provides help for any command in the application.
Simply type kubectl cobraplugin help [path to command] for full details.

Usage:
  kubectl cobraplugin help [command] [flags]

Flags:
  -h, --help   help for help
```

**Comment by nirs** [comment]
Examples run with sample-cli-plugin:
https://github.com/nirs/kubernetes/tree/sample-cli-plugin-help

```
$ kubectl ns -h
View or set the current namespace

Usage:
  kubectl ns [new-namespace] [flags]

Examples:

	# view the current namespace in your KUBECONFIG
	kubectl ns

	# view all of the namespaces in use by contexts in your KUBECONFIG
	kubectl ns --list

	# switch your current-context to one that contains the desired namespace
	kubectl ns foo


Flags:
  -h, --help                           help for kubectl ns
  ...
```

**Comment by nirs** [comment]
@jpmcb @marckhouzam did you have time to look at it?

**Review by marckhouzam — APPROVED** [review]
Thank you for following-up on these issues @nirs !
LGTM

**Pull request merged**